File: akumanomi/src/main/java/com/opmc/akumanomi/DevilFruit.java
package com.opmc.akumanomi;

/**
 * An abstract Devil Fruit
 */
public abstract class DevilFruit
{
	protected DevilFruit() { }

	/**
	 * Get this Devil Fruit's name in English
	 * @return Name, including " Fruit" suffix.
	 */
	public abstract String nameEn();

	/**
	 * Get this Devil Fruit's name in Japanese
	 * @return Name, including " no Mi" suffix.
	 */
	public abstract String nameJp();
}

File: akumanomi/src/main/kotlin/com/opmc/akumanomi/fruits/DevilFruit.kt
package com.opmc.akumanomi.fruits

/**
 * An abstract Devil Fruit, responsible for registering
 * click-combo listeners of sub-classes
 */
abstract class DevilFruit
{
	/**
	 * Get this Devil Fruit's name in English
	 * @return Name, including " Fruit" suffix.
	 */
	abstract fun nameEn(): String

	/**
	 * Get this Devil Fruit's name in Japanese
	 * @return Name, including " no Mi" suffix.
	 */
	abstract fun nameJp(): String
}